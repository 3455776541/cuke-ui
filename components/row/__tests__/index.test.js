import React from "react";
import assert from "power-assert";
import { render } from "enzyme";
import toJson from "enzyme-to-json";
import Row from "../index";
import Col from "../../col";

describe("<Row/>", () => {
	it("should render a <Row/> components", () => {
		const wrapper = render(
			<Row>
				<Col span={12} />
				<Col span={12} />
			</Row>
		);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it("should find cuke-row classnames", () => {
		const wrapper = render(
			<div>
				<Row>
					<Col span={12}>12</Col>
					<Col span={6}>6</Col>
					<Col span={2} offset={4}>
						2
					</Col>
				</Row>
			</div>
		);
		assert(wrapper.find(".cuke-row").length >= 1);
		assert(wrapper.find(".cuke-col-offset-4").length === 1);
		assert(wrapper.find(".cuke-col-6").length === 1);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
