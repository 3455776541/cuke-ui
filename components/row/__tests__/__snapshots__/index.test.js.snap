// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Row/> should find cuke-row classnames 1`] = `
<div>
  <div
    class="cuke-row"
    style="margin-left:0px;margin-right:0px"
  >
    <div
      class="cuke-col cuke-col-12"
      style="padding-left:0px;padding-right:0px"
    >
      12
    </div>
    <div
      class="cuke-col cuke-col-6"
      style="padding-left:0px;padding-right:0px"
    >
      6
    </div>
    <div
      class="cuke-col cuke-col-2 cuke-col-offset-4"
      style="padding-left:0px;padding-right:0px"
    >
      2
    </div>
  </div>
</div>
`;

exports[`<Row/> should render a <Row/> components 1`] = `
<div
  class="cuke-row"
  style="margin-left:0px;margin-right:0px"
>
  <div
    class="cuke-col cuke-col-12"
    style="padding-left:0px;padding-right:0px"
  />
  <div
    class="cuke-col cuke-col-12"
    style="padding-left:0px;padding-right:0px"
  />
</div>
`;
